### ISSUE TITLE
issues with the documentation of pal_osMutexDelete()

### ISSUE THREAD

**Issue by rgrover** [ARMmbed/pal#37]
``` C++
/*!Delete a Mutex object.
*
* @param[inout] mutexID: Mutex handle to delete, in success:(*mutexID = NULL).
*
* \return PAL_SUCCESS when mutex deleted successfully.
*         PAL_ERR_RTOS_RESOURCE - mutex already released.
*         PAL_ERR_RTOS_PARAMETER - mutex id is invalid.
*         PAL_ERR_RTOS_ISR - cannot be called from interrupt service routines.
* \note After this call the mutex_id is no longer valid and cannot be used.
*/
```
- it is not clear from the above what the platform should do if a request is made to delete a mutex which is in an acquired state. What should happen when the task which acquired a deleted mutex attempts to release it subsequently. What should happen if any tasks are blocked waiting for a mutex when it gets deleted; should such tasks be automatically unblocked?
- If mutexes can be acquired recursively (i.e. multiple times by the same thread), then there should be a way to tell when it is finally released.
- Can this API be called from an ISR context?

**Comment by teetak01** [comment]
Closing the issue as inactive and due to the repository being deprecated.

Please take a look at our current offering at https://cloud.mbed.com/docs/current. The actively maintained version of this component can be found in http://github.com/armmbed/mbed-cloud-client.

Please open a new issue if this is still valid. We strongly encourage you to migrate to the latest product. New issues in this repository will most likely remain unfixed.
